src/devices/sound/dmadac.cpp

```diff
@@ -49,11 +46,18 @@ void dmadac_sound_device::device_start()
 	save_item(NAME(m_bufout));
 	save_item(NAME(m_volume));
 	save_item(NAME(m_enabled));
-	save_item(NAME(m_frequency));
 	save_item(NAME(m_buffer));
 }
 
+void dmadac_sound_device::device_reset()
+{
+	/* reset the state */
+	m_volume = 1.0;
+	m_bufin = m_bufout = 0;
+	m_enabled = 0;
 
+	std::fill_n(m_buffer.begin(), BUFFER_SIZE, 0);
+}
```

Comment:
I’d rather this be put in some kind of public method(s) so the systems that want this behaviour can call it in their own reset methods.  There’s probably at least one system with a DMA DAC that’s free-running across system resets, and putting this in `device_reset` prevents that from working.

Reply:
Thanks, I didn't realize that would possibly be a desirable effect. I've moved the code from `device_reset` to a generic `reset` function and added the appropriate calls where required.